File: midscene_agent/src/platform.ts
/** 设备平台：与执行引擎（autoglm / midscene）解耦。 */

export type DevicePlatform = 'android' | 'harmonyos';

export type AgentBackend = 'autoglm' | 'midscene';

export function parseDevicePlatform(raw: unknown): DevicePlatform {
  const p = String(raw ?? 'harmonyos').trim().toLowerCase();
  if (p === 'android' || p === 'adb') return 'android';
  return 'harmonyos';
}

export function parseAgentBackend(raw: unknown): AgentBackend {
  const b = String(raw ?? 'midscene').trim().toLowerCase();
  return b === 'autoglm' ? 'autoglm' : 'midscene';
}

export function platformLabel(platform: DevicePlatform): string {
  return platform === 'android' ? 'Android / ADB' : 'HarmonyOS / HDC';
}

File: autoglm_phone_agent/device/platform.py
# SPDX-License-Identifier: Apache-2.0
"""设备平台类型（对齐 Open-AutoGLM --device-type adb|hdc）。"""

from __future__ import annotations

from enum import Enum


class DevicePlatform(str, Enum):
    ANDROID = "android"
    HARMONYOS = "harmonyos"

    @classmethod
    def parse(cls, raw: str | None) -> DevicePlatform:
        p = (raw or "android").strip().lower()
        if p in ("harmonyos", "harmony", "hmos", "ohos", "hdc"):
            return cls.HARMONYOS
        return cls.ANDROID

    @property
    def open_autoglm_device_type(self) -> str:
        """Open-AutoGLM CLI 参数值。"""
        return "hdc" if self == DevicePlatform.HARMONYOS else "adb"
